tmo-monitor.py

```diff
@@ -134,79 +150,217 @@ def sha256url(self, val1, val2):
+    self.parser.add_argument('--ping-count', type=int, default='1', help='how many ping health checks to perform')
+    self.parser.add_argument('--ping-interval', type=int, default='10', help='how long in seconds to wait between ping health checks')
+    #reboot settings
+    self.parser.add_argument('-R', '--reboot', action='store_true', help='skip health checks and immediately reboot gateway')
+    self.parser.add_argument('-r', '--skip-reboot', action='store_true', help='skip rebooting gateway')
+    self.parser.add_argument('--skip-bands', action='store_true', help='skip check for connected band')
+    self.parser.add_argument('--skip-5g-bands', action='store_true', help='skip check for connected 5g band')
+    self.parser.add_argument('--skip-ping', action='store_true', help='skip check for successful ping')
+    self.parser.add_argument('--skip-enbid', action='store_true', help='skip check for connected eNB ID')
+    self.parser.add_argument('--uptime', type=int, default=90, help='how long the gateway must be up before considering a reboot (defaults to 90 seconds)')
+    #connection configuration
+    self.parser.add_argument('-4', '--4g-band', type=str, action='append', dest='primary_band', default=None, choices=['B2', 'B4', 'B5', 'B12', 'B13', 'B25', 'B26', 'B41', 'B46', 'B48', 'B66', 'B71'], help='the 4g band(s) to check')
+    self.parser.add_argument('-5', '--5g-band', type=str, action='append', dest='secondary_band', default=None, choices=['n41', 'n71'], help='the 5g band(s) to check (defaults to n41)')
+    self.parser.add_argument('--enbid', type=int, default=None, help='check for a connection to a given eNB ID')
+    return self.parser.parse_args()
+
+  def parse_arguments(self, args):
+    for var in {'username', 'password'}:
+      tmp = getattr(args, var)
+      if tmp != None:
+        self.login[var] = tmp
+    for var in {'interface', 'ping_host', 'ping_count', 'ping_interval'}:
+      tmp = getattr(args, var)
+      if tmp != None:
+        self.ping[var] = tmp  
+    for var in {'primary_band', 'secondary_band', 'enbid'}:
+      tmp = getattr(args, var)
+      if tmp != None:
+        self.connection[var] = tmp
+    if args.uptime != None:
+      self.reboot['uptime'] = args.uptime
+    
+    if args.skip_ping == True:
+      self.reboot['ping'] = False
+    if self.connection['primary_band'] == '' or args.skip_bands == True:
+      self.reboot['4G_band'] = False
+    if self.connection['secondary_band'] == '' or args.skip_5g_bands == True:
+      self.reboot['5G_band'] = False
+    if self.connection['enbid'] == '' or args.skip_enbid == True:
+      self.reboot['enbid'] = False
+
+    if args.skip_reboot == True:
+      self.skip_reboot = True
+    if args.reboot == True:
+      self.reboot_now = True
+
+  def print_config(self):
+    print("Checking trashcan health:")
+    # print("  Login info:")
```

Comment:
I understand the hesitance to output credentials where the program's STDOUT might be redirected to a file. It's possible to try to detect this but this is still probably fine to leave as is, commented out.